### PR TITLE
add /api/articles endpoint with sorting and error handling, and test it

### PR DESCRIPTION
**Pull request by vikkiChipset**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -83,3 +83,39 @@ describe("GET /api/articles/:article_id", () => {
...
+          expect(article).toHaveProperty("article_id");
+          expect(article).toHaveProperty("topic");
+          expect(article).toHaveProperty("created_at");
+          expect(article).toHaveProperty("votes");
+          expect(article).toHaveProperty("article_img_url");
+          expect(article).toHaveProperty("comment_count");
+          expect(article).not.toHaveProperty("body");
+        });
+        const sortedArticles = [...articles].sort(
+          (a, b) => new Date(b.created_at) - new Date(a.created_at)
```
Comment (nc-nick-jones) [inline __tests__/app.test.js:108]: you can use jest sorted for this! it's easier.

File: app.js [hunk 2]
```diff
@@ -3,14 +3,19 @@ const app = express();
...
 const { getArticleById } = require("./controllers/articles.controller");
-const { psqlErrorHandler, customErrorHandler, serverErrorHandler } = require("./error-handling");
+const {
+  psqlErrorHandler,
+  customErrorHandler,
+  serverErrorHandler,
+} = require("./error-handling");
+const { getArticles } = require("./controllers/articles.controller");
 
 app.use(express.json());
```
Comment (nc-nick-jones) [inline app.js:13]: avoid pasting in boiler plate code - you don't need express.json() yet. please remove

**Review by nc-nick-jones — COMMENTED** [review]
a few feedback points in comments.  great work!

File: app.js (hunk 2 above)
Comment (vikkiChipset) [reply app.js:13]: I need it on the next task, can't remove

File: __tests__/app.test.js (hunk 1 above)
Comment (vikkiChipset) [reply __tests__/app.test.js:108]: got it! thanks

**Pull request merged**